### ISSUE TITLE
Passing command option leaves content in args

### ISSUE THREAD

**Issue by oliversturm** [tj/commander.js#1032]
As [discussed here](https://github.com/tj/commander.js/issues/978#issuecomment-525692913), opening as a new issue.

I'm trying to find out whether a command has been executed. The simple idea is that if a command is given, it should be executed (and nothing else). If no command is given, the default logic at the bottom should execute (but not otherwise). In all cases, the global option --load should be supported. Here's the test code:

```js
var program = require('commander');

program.version('0.0.1').option('-l, --load <name>', 'load something');

program
  .command('doit [id]')
  .description('do something')
  .option('--better', 'do it better')
  .action((id, cmd) => {
    console.log('executing doit action');
    console.log('id is', id);
    console.log('global load option is ', cmd.parent.load);
    console.log('cmd.better is', cmd.better);
  });

program.parse(process.argv);

if (program.args.length === 0)
  console.log('started without command, executing main logic');
```

This works correctly when I execute like this:

```shell
➜ node test.js -l loadname doit myid        
executing doit action
id is myid
global load option is  loadname
cmd.better is undefined
```

However, when I pass the --better option to the command, the whole thing breaks and the "main logic" is executed in addition to the command:

```shell
➜ node test.js -l loadname doit --better myid
executing doit action
id is myid
global load option is  loadname
cmd.better is true
started without command, executing main logic
```

Unexpectedly, `args` contains information after the action has been executed, *if* the command option `--better` is passed.

**Comment by shadowspawn** [comment]
(Thanks @oliversturm )

**Comment by shadowspawn** [comment]
This will be fixed by #1138

**Comment by shadowspawn** [comment]
The contents of `program.args` is now the arguments with known options removed in the published pre-release of Commander v5.0.0 (#1163)

There isn't an explicit way to directly tell if a command has been executed, but more edge cases are now handled by default (i.e. missing command, unknown command).

**Comment by shadowspawn** [comment]
Commander v5.0.0 has been released with significant changes the argument parsing.

https://github.com/tj/commander.js/releases/tag/v5.0.0
